linode/manager
# Tech Story: [M3-7947] - Update storybook to 8.0.5

## Description 📝
PR to update storybook to `8.0.5`
- See [release notes](https://storybook.js.org/blog/storybook-8/)
- See [migration guide](https://storybook.js.org/docs/migration-guide#missing-%20%20%20%E2%94%82%20%E2%94%82%20%20%20viteconfigjs-file)

I did not change anything to how we're type-checking or compiling. This could be done in subsequent optimizing PRs. It should be a 1/1 update to prepare us to adopt some extra features down the line.

I took a brief stab at adopting `react-docgen` (much faster, instead of `react-docgen-typescript`) by default since it is much faster, but type fidelity benefits were lost so best to keep the current config for now.

## Changes  🔄
- Update CM storybook to `8.0.5`
- Update other relevant packages
- Update relevant imports
- Update last remaining `.mdx` file

## Preview 📷
There should be no degradation or regression to our storybook as a result of this PR

## How to test 🧪

### Verification steps
- Pull PR locally
- Run `yarn storybook` and browse stories to confirm no regressions
- Run `yarn build-storybook` and `storybook-static` to confirm build will succeeds in our pipelines

## As an Author I have considered 🤔

*Check all that apply*
